### DISCUSSION TITLE
Controlling Coldtype Rendering Monitor in a Dual-Monitor Setup

### DISCUSSION THREAD

**Discussion by bensapirstein** [coldtype/coldtype#153] [General]
I'm currently working with a dual-monitor setup on my Mac, where the primary monitor displays the vscode Coldtype script, and the secondary monitor is dedicated to font editing software. However, I've encountered an issue where, despite having vscode open on the primary monitor, when I run Coldtype, it renders the animation on the secondary monitor.

The challenge arises when I move the rendering view back to the primary monitor; it maintains its size proportions based on the larger secondary screen, extending beyond the frame of the primary monitor. This situation makes it impractical to use the primary monitor exclusively for testing fonts with Coldtype.

Is there a way to specify or control on which monitor the rendering view appears?

**Comment by stenson** [comment]
There is! It's never worked all that well (getting monitor dimensions and pixel scales bends my mind a little), but there are some facilities for this:

`coldtype demo -mn list` should list available monitors, which you can then "select" by providing a substring name match.

For instance, I have a Dell monitor, and when I plug into that and run `-mn list`, a monitor with `DELL` in the name is printed out, so I can run: `coldtype demo -mn DELL` which theoretically should work, though I just ran that code and it looks like, for that specific monitor, there are some issues (probably monitor-dependent), which I've attempted to fix on the coldtype main branch. If you're comfortable pip installing coldtype from the git repo address, you could install the latest and try out something like this to fine-tune it, if it isn't working: `coldtype demo -mn DELL -wp SW -wpo 10,40` — the `wpo` is `window-pin-offset`, which lets you specify in a pair of ints, how many pixels the coldtype window should be offset from the `window-pin` position (here specified as `SW`). I think this will probably get into the main release of Coldtype, it's just quite difficult to debug monitor stuff on my own setup.

**Comment by bensapirstein** [comment]
Thanks that did it, I used `coldtype animations/simpleBlack.py -mn "Built-in Retina Display"` to display in the resolution of the default Mac screen
